addons/token-exchange/token_exchanger_agent.go

```diff
@@ -43,6 +47,45 @@ func NewAgentOptions() *AgentOptions {
 	return &AgentOptions{}
 }
 
+// ServeHealthProbes starts a server to check healthz and readyz probes
+func ServeHealthProbes(stop <-chan struct{}, healthProbeBindAddress string, configCheck healthz.Checker) {
+	healthzHandler := &healthz.Handler{Checks: map[string]healthz.Checker{
+		"healthz-ping": healthz.Ping,
+		"configz-ping": configCheck,
+	}}
+	readyzHandler := &healthz.Handler{Checks: map[string]healthz.Checker{
+		"readyz-ping": healthz.Ping,
+	}}
+
+	mux := http.NewServeMux()
+	mux.Handle("/readyz", http.StripPrefix("/readyz", readyzHandler))
+	mux.Handle("/healthz", http.StripPrefix("/healthz", healthzHandler))
+
+	server := http.Server{
+		Handler: mux,
+	}
+
+	ln, err := net.Listen("tcp", healthProbeBindAddress)
+	if err != nil {
+		klog.Errorf("error listening on %s: %v", ":8000", err)
```

Comment:
is it always 8000? bind-address dont have port?

Reply:
8000 is the port, it will address is not given as it will be served locally on the container.